vcr/patch.py

```diff
@@ -6,21 +6,23 @@
 from .stubs import VCRHTTPConnection, VCRHTTPSConnection
 from six.moves import http_client as httplib
 
+import logging
 
+log = logging.getLogger(__name__)
 # Save some of the original types for the purposes of unpatching
 _HTTPConnection = httplib.HTTPConnection
 _HTTPSConnection = httplib.HTTPSConnection
 
 
 # Try to save the original types for boto3
 try:
-    import botocore.vendored.requests.packages.urllib3.connectionpool as cpool
+    from botocore.awsrequest import AWSHTTPSConnection, AWSHTTPConnection
```

Comment:
I think we also need to try the `vendored` imports for users on older boto3 versions

```
try:
    from botocore.awsrequest import AWSHTTPSConnection, AWSHTTPConnection
except ImportError:
    try:
        import botocore.vendored.requests.packages.urllib3.connectionpool as cpool  
    except ImportError:  # pragma: no cover
        pass
    else:
        _Boto3VerifiedHTTPSConnection = cpool.VerifiedHTTPSConnection
        _cpoolBoto3HTTPConnection = cpool.HTTPConnection        
        _cpoolBoto3HTTPSConnection = cpool.HTTPSConnection
else:
    _Boto3VerifiedHTTPSConnection = AWSHTTPSConnection
    _cpoolBoto3HTTPConnection = AWSHTTPConnection
    _cpoolBoto3HTTPSConnection = AWSHTTPSConnection
```

Reply:
We can check for the `boto3` version and act based on that information.
Would it be good solution?
But the problem is possibly on the side of tests
which are pinned to the old version `requests27: requests==2.7.0` whereas `vendored` usually uses newer versions

Reply:
You could check for version, although it's normally better to check for behaviours rather than versions (behaviour could be reversed in a later version), hence my suggestion to use `try/except`.

Tests should probably be updated to use new requests

Reply:
Btw isn't there any better way then having all these imports on 3 places?
And implement some sort of plugin system? 
https://github.com/Rockhopper-Technologies/pluginlib

Reply:
pluginlib is cool but it may be an overkill for this